cowtown2023/src/main/java/com/team2357/frc2023/subsystems/IntakeSlideSubsystem.java

```diff
@@ -0,0 +1,156 @@
+        m_slidePIDController.setSmartMotionMinOutputVelocity(Constants.INTAKE_SLIDE.SMART_MOTION_MIN_VEL_RPM, 0);
+        m_slidePIDController.setSmartMotionMaxAccel(Constants.INTAKE_SLIDE.SMART_MOTION_MAX_ACC_RPM, 0);
+        m_slidePIDController.setSmartMotionAllowedClosedLoopError(Constants.INTAKE_SLIDE.SMART_MOTION_ALLOWED_ERROR, 0);
+    }
+
+    public void setSlideRotations(double rotations) {
+        setClosedLoopEnabled(true);
+        m_targetRotations = rotations;
+        m_slidePIDController.setReference(m_targetRotations, ControlType.kSmartMotion);
+    }
+
+    public void setSlideAxisSpeed(double axisSpeed) {
+        setClosedLoopEnabled(false);
+        double motorSpeed = (-axisSpeed) * Constants.INTAKE_SLIDE.AXIS_MAX_SPEED;
+        m_masterSlideMotor.set(motorSpeed);
+    }
+
+    public void stopSlideMotors() {
+        setClosedLoopEnabled(false);
+        m_masterSlideMotor.set(0);
+    }
+
+    public void resetEncoders() {
+        m_masterSlideMotor.getEncoder().setPosition(0);
+        m_followerSlideMotor.getEncoder().setPosition(0);
+        m_targetRotations = 0;
+    }
+
+    public double getSlideRotations() {
+        return m_masterSlideMotor.getEncoder().getPosition();
+    }
+
+    public boolean isSlideAtRotations() {
+        return Utility.isWithinTolerance(getSlideRotations(), m_targetRotations,
+                Constants.INTAKE_SLIDE.SMART_MOTION_ALLOWED_ERROR);
+    }
+
+    public boolean isExtending() {
+        return (m_desiredState == SlideState.Extended && m_currentState != SlideState.Extended);
+    }
+
+    public boolean isExtended() {
+        return (m_desiredState == SlideState.Extended && m_currentState == SlideState.Extended);
+    }
+
+    public boolean isRetracting() {
+        return (m_desiredState == SlideState.Retracted && m_currentState != SlideState.Retracted);
+    }
+
+    public boolean isRetracted() {
+        return (m_desiredState == SlideState.Retracted && m_currentState == SlideState.Retracted);
+    }
+
+    public void extend() {
+        m_currentState = SlideState.Unknown;
+        m_desiredState = SlideState.Extended;
+        setSlideRotations(Constants.INTAKE_SLIDE.SLIDE_EXTENDED_ROTATIONS);
+    }
+
+    public void retract() {
+        m_currentState = SlideState.Unknown;
+        m_desiredState = SlideState.Retracted;
+        setSlideRotations(Constants.INTAKE_SLIDE.SLIDE_RETRACTED_ROTATIONS);
+    }
+
+    @Override
+    public void periodic() {
+        if (m_currentState != m_desiredState) {
+            switch (m_desiredState) {
+                case Extended:
+                    extendPeriodic();
+                    break;
+                case Retracted:
+                    retractPeriodic();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (isClosedLoopEnabled() && isSlideAtRotations()) {
+            setClosedLoopEnabled(false);
+        }
```

Comment:
You may want to have an amp check here to make sure we are not slamming into the sides.